projects/v3/src/app/pages/v3/v3.page.ts

```diff
@@ -105,40 +106,28 @@ export class V3Page implements OnInit, OnDestroy {
     ];
   }
 
-  private _menuWithoutReview() {
-    this.appPages = [
-      {
-        title: 'Home',
-        url: '/v3/home',
-        icon: 'home'
-      },
-      {
-        title: 'Events',
-        url: '/v3/events',
-        icon: 'today'
-      },
-      {
-        title: 'Messages',
-        url: '/v3/messages',
-        icon: 'mail'
-      }
-    ];
-  }
-
   ngOnInit(): void {
-    this._menuWithoutReview();
+    this._initMenuItems();
     this.subscriptions = [];
     this.subscriptions.push(
       this.reviewService.reviews$.subscribe(res => {
         if (res && res.length) {
-          this._menuWithReview();
+          this.showReviews = true;
         } else {
-          this._menuWithoutReview();
+          this.showReviews = false;
         }
       })
     );
+
     this.subscriptions.push(this.route.params.subscribe(_params => {
       this.reviewService.getReviews();
+
+      // Hide events tab to other user roles. Show only for participants
```

Comment:
hi @sasangachathumal i prefer this logic get written outside of `this.route.params` subscription.
Writing inside `this.route.params` may give us the false sense of the `role` is a dynamic value while it isn't in our appv3, because when a user login, their role would never change until a different user login to our app.

Reply:
@trtshen I try to put it outside `this.route.params` but when it outside the `this.route.params`, check for the role not get executed when every time page load. But i will check again what i can do to improve this. Main reason i put it there `this.route.params` get executed every time page load or refreshed.